#BE-4225 Docs - Enterprise Rework (#600)
appcircleio/appcircle-docusaurus: docs/build/post-build-operations/after-a-build.md

import Screenshot from '@site/src/components/Screenshot';
import ContentRef from '@site/src/components/ContentRef';

# Build Actions

With Appcircle's post-build actions, you can easily distribute your binary file manually, access artifacts and examine build logs.

<Screenshot url='https://cdn.appcircle.io/docs/assets/build-ios-distribute-artifacts.png' />

### Distribute Binary

Distribute binary feature sends your binary file to the relevant module according to the Distribution settings in the selected configuration. 

:::caution

Please note that AAB files will not be distributed automatically since they cannot be installed on mobile devices directly.

:::

### View Build Logs

> This feature allows you to review the relevant build logs in the Appcircle log window. You can examine the logs step by step separately.

Reviewer: ```suggestion
This feature allows the relevant build logs to be reviewed in the Appcircle log window. The logs can be examined step by step separately."
```

### Download Artifacts

> Every time Appcircle completes a build, it stores all the artifacts produced after the build for easy access by the user, you can access these files by saying download artifacts. These files include artifacts such as IPA, APK and AAB binary files, log files and archive files.

Reviewer: ```suggestion
Each time a build is completed by Appcircle, all artifacts produced after the build are stored for easy access. These files, which include artifacts such as IPA, APK, and AAB binary files, log files, and archive files, can be accessed by selecting 'Download Artifacts.'
```

:::caution Output Artifacts

### For Android

If your Android application has multiple product flavors, Appcircle will create a build for each flavor of your application and let you distribute them at once. A common usage to multi-flavor applications can be free and paid versions of the same application.

When you build and distribute an application with multiple flavors, and `.apk` file will be created for each flavor. On our [**testing portal**](/testing-distribution/testing-portal), your testers will be able to download each `.apk` file separately and test it on their devices.

### For iOS

iOS applications can be downloaded or distributed as IPA file format if you configure signing identities and sign the application during the build process.

If you disable signing or don't use any signing identities, iOS output will be an `.xarchive` file.

:::

### Download Build Logs

> In addition to allowing you to review logs, Appcircle also allows you to download these logs in .txt file format, so you can download and use your logs in a single file.

Reviewer: ```suggestion
In addition to allowing the review of logs, Appcircle also permits the downloading of these logs in .txt file format, enabling the logs to be downloaded and used in a single file.
```
